lib/src/realm_class.dart

```diff
@@ -539,6 +539,21 @@ class Realm implements Finalizable {
       realm.close();
     }
   }
+
+  /// Update the `Realm` instance and outstanding objects to point to the most recent persisted version.
+  /// 
+  /// If another process or thread has made changes to the realm file, this causes
+  /// those changes to become visible in this realm instance.
+  /// Typically you don't need to call this method since Realm has auto-refresh built-in.
+  bool refresh() {
+    return realmCore.realmRefresh(this);
+  }
+
+  /// Returns a [Future] that will complete the first time the realm is refreshed to the version which is the
+  /// latest version at the time when this method is called.
+  Future<void> onRefresh(void Function() action) {
+    return realmCore.onRealmRefresh(this).whenComplete(action);
+  }
```

Comment:
Why is this taking an action? I'd imagine an API like `Future<void> refreshAsync()` and if you want to do something after the refresh you use:

```
await realm.refreshAsync();
action();
```

It seems redundant to be exposing this as both a promise and a method that takes a callback.

Reply:
This is not refreshAsync implementation but rather a add_refresh_callback implementation. This takes a callback which will get executed only once and returns a Future since you might want to await it or most probably not. Just register the callback.